### PR TITLE
Col plan multilayer select

### PR DESCRIPTION
**Pull request by thebioengineer**
addresses #113 , and adds additional behavior to handle `-` in col_plans when the field is renamed.

Adds tests for these cases as well.

It was found that with _multiple_ levels of spanning headers, there were differences in how the names were created for matching in select_col_plan and clean_spanning_col_names. these are now in line.

### CODE REVIEW

**Comment by thebioengineer** [comment]
tbh not sure how to correct the failing runs. I'll look into how I can get those environments and test in them

File: R/col_plan.R [hunk 1]
```diff
@@ -343,27 +355,47 @@ select_col_plan <- function(data, tfrmt){
...
-          new_name_in_df = case_when(
-            new_name_in_df == .original_col ~ NA_character_,
-            TRUE ~ new_name_in_df
-          )
-        )
+          new_name_in_df = remove_empty_layers(.data$new_name_in_df, n_layers),
+          new_name_in_df_output = remove_empty_layers(.data$new_name_in_df_output, n_layers)
+        ) %>%
+        mutate(
+          new_name_quo = map2(.data$new_name_in_df, .data$.removal_identifier_col, function(x, y) {
```
Comment (statasaurus) [inline R/col_plan.R:365]: Given this already an anonymous function can you make it a named function and pull it out this easier to maintain

File: R/col_plan.R [hunk 2]
```diff
@@ -343,27 +355,47 @@ select_col_plan <- function(data, tfrmt){
...
+            x_text <- tryCatch({
+              x_lang <- parse(text = x)[[1]]
+              if (is_valid_tidyselect_call(x_lang)) {
+                x
+              } else{
+                paste0("`", x, "`")
```
Comment (statasaurus) [inline R/col_plan.R:371]: Can you add additional comments here?

**Review by statasaurus — APPROVED** [review]
My comments here are more take it or leave it.

File: tests/testthat/test-col_plan.R [hunk 3]
```diff
@@ -457,6 +457,161 @@ test_that("Build simple tfrmt with multiple columns and with renaming duplicated
...
+      col1,
+      col2,
+      col4,
+      new_col_3 = mycol3,
+      mycol5
+    )
+  )
+
+
+  suppressMessages({
```
[the remark this answers is not in the document]
Comment (thebioengineer) [reply tests/testthat/test-col_plan.R:523]: TBH I don't recall why I was doing the warning suppression here. We don't want the messages b/c its the message which rows don't get processed by the body plan. We may want to rethink that a bit if there was no body_plan provided whether to return that message.

File: R/apply_tfrmt.R [hunk 4]
```diff
@@ -71,12 +71,17 @@ tentative_process <- function(.data, fx, ..., fail_desc = NULL){
...
       safely(fx)(...)
     if(!is.null(out[["error"]])){
-      out <- .data
       if(is.null(fail_desc)){
-        message("Unable to to apply formatting", format(substitute(fx)))
-      }else{
-        message(fail_desc)
+        fail_desc <- paste0("Unable to to apply ",format(substitute(fx)),".")
       }
+      fail_desc <-paste0(
```
[the remark this answers is not in the document]
Comment (thebioengineer) [reply R/apply_tfrmt.R:77]: NP. I was using it to debug too. So figured we should keep that.

File: R/col_plan.R (hunk 1 above)
Comment (thebioengineer) [reply R/col_plan.R:365]: NP. will do.

**Pull request merged**